src/main/java/io/confluent/connect/jdbc/validation/JdbcSourceConnectorValidation.java

```diff
@@ -0,0 +1,440 @@
+    }
+    
+    List<String> timestampColumnsMapping = config.getTimestampColumnMapping();
+    if (timestampColumnsMapping != null && !timestampColumnsMapping.isEmpty()) {
+      addConfigError(JdbcSourceConnectorConfig.TIMESTAMP_COLUMN_MAPPING_CONFIG, msg);
+    }
+    
+    String incrementingColumnName = config.getIncrementingColumnName();
+    if (incrementingColumnName != null && !incrementingColumnName.trim().isEmpty()) {
+      addConfigError(JdbcSourceConnectorConfig.INCREMENTING_COLUMN_NAME_CONFIG, msg);
+    }
+    
+    List<String> incrementingColumnMapping = config.getIncrementingColumnMapping();
+    if (incrementingColumnMapping != null && !incrementingColumnMapping.isEmpty()) {
+      addConfigError(JdbcSourceConnectorConfig.INCREMENTING_COLUMN_MAPPING_CONFIG, msg);
+    }
+    
+    return false;
+  }
+  
+  /**
+   * Validate that at least one configuration is provided.
+   */
+  private boolean addConfigErrorsForNoConfigProvided() {
+    String msg = "At least one table filtering configuration is required. "
+        + "Provide one of: " + JdbcSourceConnectorConfig.TABLE_WHITELIST_CONFIG + ", "
+        + JdbcSourceConnectorConfig.TABLE_BLACKLIST_CONFIG + ", "
+        + JdbcSourceConnectorConfig.TABLE_INCLUDE_LIST_CONFIG + ", or "
+        + JdbcSourceConnectorConfig.TABLE_EXCLUDE_LIST_CONFIG + ".";
+    
+    addConfigError(JdbcSourceConnectorConfig.TABLE_WHITELIST_CONFIG, msg);
+    addConfigError(JdbcSourceConnectorConfig.TABLE_BLACKLIST_CONFIG, msg);
+    addConfigError(JdbcSourceConnectorConfig.TABLE_INCLUDE_LIST_CONFIG, msg);
+    addConfigError(JdbcSourceConnectorConfig.TABLE_EXCLUDE_LIST_CONFIG, msg);
+    return false;
+  }
+  
+  /**
+   * Validate new config requirements (when using new configs only).
+   */
+  private boolean validateNewConfigRequirements() {
+    Set<String> includeListSet = config.getTableIncludeListSet();
+    Set<String> excludeListSet = config.getTableExcludeListSet();
+    
+    // Validate that exclude list requires include list
+    if (excludeListSet != null && includeListSet == null) {
+      String msg = JdbcSourceConnectorConfig.TABLE_EXCLUDE_LIST_CONFIG 
+          + " cannot be used without " + JdbcSourceConnectorConfig.TABLE_INCLUDE_LIST_CONFIG
+          + ". Exclude list only applies to tables that match the include list.";
+      addConfigError(JdbcSourceConnectorConfig.TABLE_EXCLUDE_LIST_CONFIG, msg);
+      return false;
+    }
+    
+    return true;
+  }
+
+  /**
+   * Validate plugin-specific needs. This method can be overridden by specific
+   * connector implementations to add their own validation logic.
+   */
+  protected boolean validatePluginSpecificNeeds() {
+    // Default implementation - no plugin-specific validation
+    // This can be overridden by specific connector implementations
+    return true;
+  }
+
+  /**
+   * Validate that mode-dependent column configurations are properly provided.
+   */
+  private boolean validateTsAndIncModeColumnRequirements() {
```

Comment:
Currently the functions in this also check for legacy configs. Let's remove that. We are only validating new configs. And if validation reaches here, it's definitely only new configs present.